android/src/main/java/com/guichaguri/trackplayer/service/MusicBinder.java

```diff
@@ -16,6 +18,8 @@
     public MusicBinder(MusicService service, MusicManager manager) {
         this.service = service;
         this.manager = manager;
+        
+        IcyStreamUpdate.setService(service);
```

Comment:
Here's why you shouldn't make it a static class: You're leaking memory. You're storing the service in a field, but not removing it later.
That's why all the classes used by the service are instances, so they can be removed by the garbage collector when the service stops.